src/components/EditFreakForm/__test__/EditFreakForm.test.jsx

```diff
@@ -0,0 +1,222 @@
+
+    const roleInput = screen.queryByTestId('role-input');
+    const roleOptions = screen.queryAllByTestId('role-option');
+    const firstOptionText = roleOptions[0].textContent;
+    fireEvent.change(roleInput, { target: { value: firstOptionText } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.role]: firstOptionText,
+    });
+  });
+
+  it('should trigger onChange when the level is changed with one of the options', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const levelInput = screen.queryByTestId('level-input');
+    const levelOptions = screen.queryAllByTestId('level-option');
+    const fifthOptionText = levelOptions[5].textContent;
+
+    fireEvent.change(levelInput, { target: { value: fifthOptionText } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.level]: fifthOptionText,
+    });
+  });
+
+  it('should trigger onChange when the level is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const levelInput = screen.getByTestId('level-input');
+
+    fireEvent.change(levelInput, { target: { value: 'Novice' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.level]: 'Novice',
+    });
+  });
+
+  it('should trigger onChange when the norm is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const normInput = screen.queryByTestId('norm-input');
+    const normOptions = screen.queryAllByTestId('norm-option');
+    const thirdOptionText = normOptions[1].textContent;
+
+    fireEvent.change(normInput, { target: { value: thirdOptionText } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.norm]: thirdOptionText,
+    });
+  });
+
+  it('should trigger onChange when the skills is changed', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const skillsInput = screen.getByTestId('skills-input');
+
+    fireEvent.change(skillsInput, { target: { value: 'Elm' } });
+
+    expect(onChange).toBeCalledWith({
+      ...FreakModelDefault,
+      [FreakModelKeys.skills]: 'Elm',
+    });
+  });
+
+  it('should receive a freak and display its data', () => {
+    render(
+      <EditFreakForm
+        freak={ freak }
+        onChange={ onChange }
+      />,
+    );
+
+    const lastNameInput = screen.getByTestId('last-name-input');
+    expect(lastNameInput.value).toEqual(freak.lastName);
+  });
+
+  it('should display default data if not given any data', () => {
+    render(
+      <EditFreakForm
+        onChange={ onChange }
+      />,
+    );
+
+    const lastNameInput = screen.getByTestId('last-name-input');
+    expect(lastNameInput.value).toEqual('');
+  });
+
+  it('should trigger onChange when the firstName is given some data', () => {
```

Comment:
This description does not sound correct. The onChange is triggered when the user changes something, and it says that is triggered when there is some data.

What about 'should trigger onChange when there is an update, and it gives the entry data with the new change '